code/modules/client/preference/preferences_mysql.dm

```diff
@@ -495,3 +501,20 @@
 	load_character(C,pick(saves))
 	return 1
 
+/datum/preferences/proc/clear_character_slot(client/C)
+	. = FALSE
+	// Is there a character in that slot?
+	var/DBQuery/query = GLOB.dbcon.NewQuery("SELECT slot FROM [format_table_name("characters")] WHERE ckey='[C.ckey]' AND slot='[default_slot]'")
+	query.Execute()
+	if(!query.RowCount())
+		return
+
+	query = GLOB.dbcon.NewQuery("DELETE FROM [format_table_name("characters")] WHERE ckey='[C.ckey]' AND slot='[default_slot]'")
```

Comment:
Dont re-use the same query object. Make a new one.

Reply:
Donezo